### PR TITLE
[WIP] Implement Proposals.HSlice

### PR DESCRIPTION
**Pull request by SaranjeetKaur**

### CODE REVIEW

File: src/proposals/Proposals.jl [hunk 1]
```diff
@@ -1,14 +1,13 @@
 """
     NestedSamplers.Proposals
-
 This module contains the different algorithms for proposing new points within a bounding volume in unit space.
-
```
Comment (mileslucas) [inline src/proposals/Proposals.jl:5]: I'm not sure why you've removed this white space? Please revert this.

Comment (SaranjeetKaur) [reply src/proposals/Proposals.jl:5]: I am sorry, I really don't remember when exactly did I do this! I will revert it.

Comment (SaranjeetKaur) [reply src/proposals/Proposals.jl]: Done!

**Comment by cpfiffer** [comment]
Also, all the tests are failing. Definitely a good idea to fix those before we can make any meaningful comments.

**Comment by mileslucas** [comment]
> Also, this is a general question for this repo, how is the actual accuracy of these proposals being tested? Like if I give it a simple model, does it get the mean/variance right?

Right now I'm testing some trivial properties of the proposals, such as "does the proposed point have higher likelihood than the critical value?" Implicitly it also makes sure we don't hit any of those errors related to extremely inefficient sampling (this is all in `test/proposals.jl`). In addition, they're being used inside the sampling tests, which _do_ have analytical outputs we can test against (this is in `test/sampling.jl`).

Looking at other nested sampling libraries this actually seems like more coverage already. For example, dynesty's tests are almost all integration tests that don't even test all combos of proposals and bounds.

I'd love to get some more granularity to be able to test the corner cases and expectations of the proposals, but this work isn't directly part of my current research so I can't commit as much time to this package to polish off the corners.
